test/dd/test_dd_functionality.cpp

```diff
@@ -172,6 +177,41 @@ TEST_P(DDFunctionality, controlledStandardOpBuildInverseBuild) {
   dd->incRef(e);
 
   EXPECT_EQ(ident, e);
+
+  if (qc::isTwoQubitGate(gate)) {
+    ASSERT_NO_THROW(
+        { d = dd->multiply(getDD(&op_neg, *dd), getInverseDD(&op_neg, *dd)); });
+    dd->incRef(d);
+
+    EXPECT_EQ(ident, d);
+  }
+}
+
+TEST_F(DDFunctionality, twoTargetControlledGateDDEdgeCases) {
+  using namespace qc::literals;
+  auto newNQubits = 12U;
+  auto newDD = std::make_unique<dd::Package<>>(newNQubits);
+  auto gate = SWAP;
+
+  qc::StandardOperation op, op_neg;
+  op = qc::StandardOperation(newNQubits, Controls{1, 5, 7, 11}, 3, 9, gate);
+  op_neg = qc::StandardOperation(newNQubits, Controls{1_nc, 5_nc, 7_nc, 11_nc},
+                                 3, 9, gate);
+
+  ASSERT_NO_THROW(
+      { e = newDD->multiply(getDD(&op, *newDD), getInverseDD(&op, *newDD)); });
+  newDD->incRef(e);
+
+  ASSERT_NO_THROW({
+    d = newDD->multiply(getDD(&op_neg, *newDD), getInverseDD(&op_neg, *newDD));
+  });
+  newDD->incRef(d);
+
+  // the DD of the identity needs to be reconstructed as well
+  ident = newDD->makeIdent(newNQubits);
+
+  EXPECT_EQ(ident, e);
+  EXPECT_EQ(ident, d);
```

Comment:
This test seems oddly specific to me. What part of that is not already covered by existing tests?
If it's the part about the negative controls, then this should be fairly easy to incorporate into the existing tests that test out all the possibilities of combining controls and targets by adding another for loop that influences the type of control.